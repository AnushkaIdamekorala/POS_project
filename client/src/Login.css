@import url("https://fonts.googleapis.com/css?family=Numans");

body {
  background-image: url("https://cdn.asiatatler.com/asiatatler/i/hk/2019/01/11160759-seafood-selection-dinner_article_1499x1000.jpg");
  background-size: cover;
  background-repeat: no-repeat;
  height: 100%;
  font-family: "Numans", sans-serif;
}

.container {
  height: 100%;
  align-content: center;
}

.cardd {
  height: 370px;
  margin-top: auto;
  margin-bottom: auto;
  width: 400px;
  background-color: rgba(0, 0, 0, 0.5) !important;
}

.social_icon span {
  font-size: 60px;
  margin-left: 10px;
  color: #ffc312;
}

.social_icon span:hover {
  color: white;
  cursor: pointer;
}

.card-header h3 {
  color: white;
}

.social_icon {
  position: absolute;
  right: 20px;
  top: -45px;
}

.input-group-prepend span {
  width: 50px;
  background-color: #dea1e0;
  color: black;
  border: 0 !important;
}

input:focus {
  outline: 0 0 0 0 !important;
  box-shadow: 0 0 0 0 !important;
}

.remember {
  color: white;
}

.remember input {
  width: 20px;
  height: 20px;
  margin-left: 15px;
  margin-right: 5px;
}

.login_btn {
  color: black;
  background-color: #8cd8a3;
  width: 100px;
}

.login_btn:hover {
  color: black;
  background-color: white;
}

.links {
  color: white;
}

.links a {
  margin-left: 4px;
}

.madata {
  margin: 0;
  position: absolute;
  top: 30%;
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}
