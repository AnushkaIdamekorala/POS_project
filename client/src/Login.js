import React, { Component } from "react";
import WarningModal from "./components/WarningModal";
import { Link } from "react-router-dom";

import "bootstrap/dist/css/bootstrap.min.css";
import "./Login.css";

import { loginUser, userCarts } from "./actions/userActions";
import { connect } from "react-redux";
import PropTypes from "prop-types";

class Login extends Component {
  state = {
    email: "",
    password: ""
  };

  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  onSubmit = e => {
    e.preventDefault();
    const userDetail = {
      email: this.state.email,
      password: this.state.password
    };
    this.props.loginUser(userDetail);
  };

  onWord = () => {
    this.props.history.push("");
  };

  componentDidMount() {
    if (this.props.auth.isAuthenticated === true) {
      this.props.history.push("/");
    }
  }

  render() {
    if (this.props.auth.isAuthenticated === true) {
      this.props.history.push("/");
      return null;
    } else {
      document.body.style.backgroundImage =
        "url('https://image.shutterstock.com/image-photo/cheerful-mature-women-enjoying-funny-260nw-640975693.jpg')";
      return (
        <div class="container madata">
          <WarningModal />
          <div class="d-flex justify-content-center h-100 ">
            <div class="cardd">
              <div class="card-header">
                <h3>Sign In</h3>
              </div>
              <div class="card-body">
                <form>
                  <div class="input-group form-group">
                    <div class="input-group-prepend">
                      <span class="input-group-text">
                        <i class="fas fa-user" />
                      </span>
                    </div>
                    <input
                      type="text"
                      class="form-control"
                      name="email"
                      placeholder="email"
                      onChange={this.onChange}
                    />
                  </div>
                  <div class="input-group form-group">
                    <div class="input-group-prepend">
                      <span class="input-group-text">
                        <i class="fas fa-key" />
                      </span>
                    </div>
                    <input
                      type="password"
                      class="form-control"
                      name="password"
                      placeholder="password"
                      onChange={this.onChange}
                    />
                  </div>
                  <div class="row align-items-center remember">
                    <input type="checkbox" />
                    Remember Me
                  </div>
                  <div class="form-group">
                    <input
                      type="submit"
                      value="Login"
                      class="btn float-right login_btn"
                      onClick={this.onSubmit}
                    />
                  </div>
                </form>
              </div>
              <div class="card-footer">
                <div class="d-flex justify-content-center links">
                  Don't have an account?
                  <Link to="/signup">
                    <font color="blue">Login</font>
                  </Link>
                </div>
              </div>
            </div>
          </div>
        </div>
      );
    }
  }
}

Login.propTypes = {
  auth: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
  { loginUser, userCarts }
)(Login);
