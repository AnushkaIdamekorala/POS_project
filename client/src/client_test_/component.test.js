import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Login from "../Login";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import sinon from "sinon";
import { BrowserRouter as Router } from "react-router-dom";

import configureStore from "redux-mock-store";

const middlewares = [thunk];

Enzyme.configure({ adapter: new Adapter() });

describe("components", () => {
  describe("Login component", () => {
    const mockStore = configureStore(middlewares);
    let store;
    const loginUser = sinon.spy();
    const userCarts = sinon.spy();

    const initialState = {
      auth: {
        email: "",
        password: ""
      },
      loginUser,
      userCarts
    };

    beforeEach(() => {
      store = mockStore(initialState);
    });
    it("should render self and subcomponents", () => {
      /* const props = {
        login: jest.fn()
      }; */
      const wrapper = mount(
        <Provider store={store}>
          <Router>
            <Login />
          </Router>
        </Provider>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it("simulate button action", () => {
      const wrapper = mount(
        <Provider store={store}>
          <Router>
            <Login />
          </Router>
        </Provider>
      );
      //console.log(wrapper.debug());
      wrapper.find(".login_btn").simulate("click");
      // expect(loginUser.calledOnce).toBe(true);
      //   expect(loginUser.calledOnce);
      expect(wrapper.find(".form-control").simulate("change"));
    });
  });
});
