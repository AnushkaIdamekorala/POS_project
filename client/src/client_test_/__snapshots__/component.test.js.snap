// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Login component should render self and subcomponents 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Login)>
        <Login
          auth={
            Object {
              "email": "",
              "password": "",
            }
          }
          loginUser={[Function]}
          userCarts={[Function]}
        >
          <div
            className="container madata"
          >
            <Connect(WarningModal)>
              <WarningModal
                auth={
                  Object {
                    "email": "",
                    "password": "",
                  }
                }
                userRefresh={[Function]}
              >
                <div>
                  <Modal
                    autoFocus={true}
                    backdrop={true}
                    backdropTransition={
                      Object {
                        "mountOnEnter": true,
                        "timeout": 150,
                      }
                    }
                    centered={false}
                    fade={true}
                    isOpen={false}
                    keyboard={true}
                    modalTransition={
                      Object {
                        "timeout": 300,
                      }
                    }
                    onClosed={[Function]}
                    onOpened={[Function]}
                    role="dialog"
                    toggle={[Function]}
                    zIndex={1050}
                  />
                </div>
              </WarningModal>
            </Connect(WarningModal)>
            <div
              className="d-flex justify-content-center h-100 "
            >
              <div
                className="cardd"
              >
                <div
                  className="card-header"
                >
                  <h3>
                    Sign In
                  </h3>
                </div>
                <div
                  className="card-body"
                >
                  <form>
                    <div
                      className="input-group form-group"
                    >
                      <div
                        className="input-group-prepend"
                      >
                        <span
                          className="input-group-text"
                        >
                          <i
                            className="fas fa-user"
                          />
                        </span>
                      </div>
                      <input
                        className="form-control"
                        name="email"
                        onChange={[Function]}
                        placeholder="email"
                        type="text"
                      />
                    </div>
                    <div
                      className="input-group form-group"
                    >
                      <div
                        className="input-group-prepend"
                      >
                        <span
                          className="input-group-text"
                        >
                          <i
                            className="fas fa-key"
                          />
                        </span>
                      </div>
                      <input
                        className="form-control"
                        name="password"
                        onChange={[Function]}
                        placeholder="password"
                        type="password"
                      />
                    </div>
                    <div
                      className="row align-items-center remember"
                    >
                      <input
                        type="checkbox"
                      />
                      Remember Me
                    </div>
                    <div
                      className="form-group"
                    >
                      <input
                        className="btn float-right login_btn"
                        onClick={[Function]}
                        type="submit"
                        value="Login"
                      />
                    </div>
                  </form>
                </div>
                <div
                  className="card-footer"
                >
                  <div
                    className="d-flex justify-content-center links"
                  >
                    Don't have an account?
                    <Link
                      replace={false}
                      to="/signup"
                    >
                      <a
                        href="/signup"
                        onClick={[Function]}
                      >
                        <font
                          color="blue"
                        >
                          SignUp
                        </font>
                      </a>
                    </Link>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Login>
      </Connect(Login)>
    </Router>
  </BrowserRouter>
</Provider>
`;
